src/Controller/JobController.php

```diff
@@ -0,0 +1,327 @@
+    {
+        try {
+            $this->jobValidator->nameIsValid($name);
+            $rows = $this->jobsRepository->selectByName($name);
+            $jobsArr = [];
+            foreach ($rows as $job) {
+                $jobsArr[] = [
+                    'id' => $job->getId(),
+                    'name' => $job->getName(),
+                    'description' => $job->getDescription(),
+                    'createdAt' => $job->getCreatedAt(),
+                    'company' => [
+                        'id' => $job->getCompany()->getId(),
+                        'company' => $job->getCompany()->getName()
+                        
+                    ]
+                ];
+            }
+
+            return new JsonResponse([
+                'results' => [
+                    'error' =>false,
+                    'jobs' => $jobsArr
+                ]
+            ]);
+        } catch (\Exception $e) {
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'error' => true,
+                        'message' => $e->getMessage()
+                    ]
+                ]
+            );
+        }
+    }
+  
+    /**
+     * read a job by the substring of the name
+     *
+     * @param  mixed $name
+     * @return Response
+     */
+    public function readJobsByNameLike(string $name): Response
+    {
+        try {
+            $this->jobValidator->nameIsValid($name);
+            $rows = $this->jobsRepository->selectByNameLike($name);
+            $jobsArr = [];
+            foreach ($rows as $job) {
+                $jobsArr[] = [
+                    'id' => $job->getId(),
+                    'name' => $job->getName(),
+                    'description' => $job->getDescription(),
+                    'createdAt' => $job->getCreatedAt(),
+                    'company' => [
+                        'id' => $job->getCompany()->getId(),
+                        'company' => $job->getCompany()->getName()
+                        
+                    ]
+                ];
+            }
+            
+            return new JsonResponse([
+                'results' => [
+                    'error' =>false,
+                    'jobs' => $jobsArr
+                ]
+            ]);
+        } catch (\Exception $e) {
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'error' => true,
+                        'message' => $e->getMessage()
+                    ]
+                ]
+            );
+        }
+    }
+
+    /**
+     * updates a job from the database
+     *
+     * @param  mixed $id
+     * @param  mixed $request
+     * @return Response
+     */
+    public function updateJob(int $id, Request $request): Response
+    {
+        try {
+            $this->jobValidator->idIsValid($id);
+            $name = $request->get('name');
+            $this->jobValidator->nameIsValid($name);
```

Comment:
name is not mandatory, do the validation only if request parameter is sent by the request.